code/modules/power/antimatter/control.dm

```diff
@@ -268,92 +271,117 @@
-	dat += "Status: [(active?"Injecting":"Standby")] <BR>"
-	dat += "<A href='?src=[REF(src)];togglestatus=1'>Toggle Status</A><BR>"
-
-	dat += "Stability: [stability]%<BR>"
-	dat += "Reactor parts: [linked_shielding.len]<BR>"//TODO: perhaps add some sort of stability check
-	dat += "Cores: [linked_cores.len]<BR><BR>"
-	dat += "-Current Efficiency: [reported_core_efficiency]<BR>"
-	dat += "-Average Stability: [stored_core_stability] <A href='?src=[REF(src)];refreshstability=1'>(update)</A><BR>"
-	dat += "Last Produced: [DisplayPower(stored_power)]<BR>"
-
-	dat += "Fuel: "
-	if(!fueljar)
-		dat += "<BR>No fuel receptacle detected."
-	else
-		dat += "<A href='?src=[REF(src)];ejectjar=1'>Eject</A><BR>"
-		dat += "- [fueljar.fuel]/[fueljar.fuel_max] Units<BR>"
-
-		dat += "- Injecting: [fuel_injection] units<BR>"
-		dat += "- <A href='?src=[REF(src)];strengthdown=1'>--</A>|<A href='?src=[REF(src)];strengthup=1'>++</A><BR><BR>"
-
-
-	user << browse(dat, "window=AMcontrol;size=420x500")
-	onclose(user, "AMcontrol")
-	return
+	ShowInterface(user)
 
 
 /obj/machinery/power/am_control_unit/Topic(href, href_list)
 	if(..())
-		return
+		return TRUE
 
 	if(href_list["close"])
-		usr << browse(null, "window=AMcontrol")
-		usr.unset_machine()
-		return
+		if(usr.machine == src)
+			usr.unset_machine()
+		return TRUE
 
-	if(href_list["togglestatus"])
-		toggle_power()
-
-	if(href_list["refreshicons"])
-		update_shield_icons = 1
-
-	if(href_list["ejectjar"])
-		if(fueljar)
-			fueljar.forceMove(drop_location())
+	//Ignore input if we are broken or guy is not touching us, AI can control from a ways away
+	if(stat & (BROKEN|NOPOWER))
+		usr.unset_machine()
+		usr << browse(null, "window=AMcontrol")
+		return TRUE
+
+	switch(href_list["task"])
+		if("close")
+			usr << browse(null, "window=AMcontrol")
+			usr.unset_machine()
+		if("togglestatus")
+			toggle_power()
+			message_admins("AME toggled [active?"on":"off"] by [usr.real_name] ([key_name_admin(usr)]) at ([COORD(loc)] - [ADMIN_JMP(loc)]",0,1)
+		if("eject")
+			if(fueljar)
+				fueljar.forceMove(drop_location())
 			fueljar = null
-			//fueljar.control_unit = null currently it does not care where it is
-			//update_icon() when we have the icon for it
-
-	if(href_list["strengthup"])
-		fuel_injection++
-
-	if(href_list["strengthdown"])
-		fuel_injection--
-		if(fuel_injection < 0)
-			fuel_injection = 0
-
-	if(href_list["refreshstability"])
-		check_core_stability()
+		if("set_strength")
+			var/newval = input("Enter new injection strength") as num|null
+			if(!newval)
+				return
+			fuel_injection = max(1, newval)
+			message_admins("AME injection strength set to [fuel_injection] by [usr.real_name] ([key_name_admin(usr)]) at ([COORD(loc)] - [ADMIN_JMP(loc)])",FALSE,TRUE)
+		if("refreshicons")
+			check_core_stability()
 
 	updateDialog()
 	return
+
+/obj/machinery/power/am_control_unit/proc/ShowInterface(mob/user)
+	if(!user || !(in_range(src, user) || issilicon(user)))

```

Comment:
Doesn't the silicon check here make is so that silicon's can't use the UI?
Seems unintentional with the checks below.